client-course-schedulizer/src/components/App/App.tsx

```diff
@@ -13,6 +21,68 @@ export const App = () => {
   const [isCSVLoading, setIsCSVLoading] = useState(false);
   const { save } = useLocal("appState");
   const [appState, appDispatch] = useReducer(reducer(save), initialAppState);
+  const { schedule, fileUrl } = appState;
+
+  // Load a file from the GET parameters (either ?csv= or ?xlsx=),
+  // only do this once for each page refresh and only do this if the initial state is empty
+  // Ref: https://stackoverflow.com/questions/56663785/invalid-hook-call-hooks-can-only-be-called-inside-of-the-body-of-a-function-com
+  // Ref: https://www.gitmemory.com/issue/SheetJS/js-xlsx/1110/531003740
+  useEffect(() => {
+    const scheduleHasCourses = schedule.courses.length > 0;
+    if (!scheduleHasCourses) {
+      // eslint-disable-next-line no-restricted-globals
+      const url = location.href;
+      // TODO: We'll need to be more careful about this if we end up using more GET params
+      const csvIndex = url.indexOf("?csv=");
+      const xlsxIndex = url.indexOf("?xlsx=");
+      if (csvIndex !== -1) {
+        const csvUrl = url.slice(csvIndex + 5);
```

Comment:
What is this number's importance? I would suggest making it a constant

Reply:
Oh, I guess I should make a `"?csv="` a variable and use `.length`

Reply:
893fc3c